### ISSUE TITLE
Provide pre-release name via required branch

### ISSUE THREAD

**Issue by ivandov** [release-it/release-it#811]
When configuring a `requireBranch`, it'd be nice if this key supported an object format that allowed you to specify multiple branches, but also information on whether that branch is a pre-release branch. 

For example:
```
requireBranch:
  - name: v2.1.0-alpha
    preRelease: alpha
```

Therefore when `release-it` runs on the `v2.1.0-alpha` branch, it will always release `pre-release` semver types with the name `alpha`... `v2.1.0-alpha.X`

Currently, I believe any of this additional logic may need to be handled in a `hook` if not a direct `plugin` as it would need to overwrite the determined version in the `bump` logic.

Trying to find a solution similar to `semantic-release` [pre-release branches](https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/pre-releases.md).

**Comment by webpro** [comment]
FYI, this just landed: #877

**Comment by ivandov** [comment]
Is there a way to use the support for multi-branches that landed in #877 to label specific branches as being pre-release branches?
